wdl-ast/src/element.rs

```diff
@@ -0,0 +1,755 @@
+    Token(Token),
+}
+
+impl Element {
+    /// Attempts to get a reference to the inner [`Node`].
+    ///
+    /// * If `self` is a [`Element::Node`], then a reference to the inner
+    ///   [`Node`] wrapped in [`Some`] is returned.
+    /// * Else, [`None`] is returned.
+    pub fn as_node(&self) -> Option<&Node> {
+        match self {
+            Self::Node(node) => Some(node),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner [`Node`].
+    ///
+    /// * If `self` is a [`Element::Node`], then the inner [`Node`] wrapped in
+    ///   [`Some`] is returned.
+    /// * Else, [`None`] is returned.
+    pub fn into_node(self) -> Option<Node> {
+        match self {
+            Self::Node(node) => Some(node),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner [`Node`].
+    ///
+    /// # Panics
+    ///
+    /// If `self` is not a [`Element::Node`].
+    pub fn unwrap_node(self) -> Node {
+        self.into_node()
+            .expect("expected `Element::Node` but got a different variant")
+    }
+
+    /// Attempts to get a reference to the inner [`Token`].
+    ///
+    /// * If `self` is a [`Element::Token`], then a reference to the inner
+    ///   [`Token`] wrapped in [`Some`] is returned.
+    /// * Else, [`None`] is returned.
+    pub fn as_token(&self) -> Option<&Token> {
+        match self {
+            Self::Token(token) => Some(token),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and attempts to return the inner [`Token`].
+    ///
+    /// * If `self` is a [`Element::Token`], then the inner [`Token`] wrapped in
+    ///   [`Some`] is returned.
+    /// * Else, [`None`] is returned.
+    pub fn into_token(self) -> Option<Token> {
+        match self {
+            Self::Token(token) => Some(token),
+            _ => None,
+        }
+    }
+
+    /// Consumes `self` and returns the inner [`Token`].
+    ///
+    /// # Panics
+    ///
+    /// If `self` is not a [`Element::Token`].
+    pub fn unwrap_token(self) -> Token {
+        self.into_token()
+            .expect("expected `Element::Token` but got a different variant")
+    }
+
+    /// Gets the underlying [`SyntaxElement`] from the [`Element`].
+    pub fn syntax(&self) -> SyntaxElement {
+        match self {
+            Element::Node(node) => SyntaxElement::Node(node.syntax().clone()),
+            Element::Token(token) => SyntaxElement::Token(token.syntax().clone()),
+        }
+    }
+
+    /// Gets the underlying [`SyntaxKind`] from the [`Element`].
+    pub fn kind(&self) -> SyntaxKind {
+        match self {
+            Element::Node(node) => node.syntax().kind(),
+            Element::Token(token) => token.syntax().kind(),
+        }
+    }
+
+    /// Returns whether the [`SyntaxElement`] represents trivia.
+    pub fn is_trivia(&self) -> bool {
+        match self {
+            Element::Node(node) => node.syntax().kind().is_trivia(),
```

Comment:
What's an example of a `Node` that is `is_trivia() == true`? I would think that by nature of something being a `Node`, it can just respond `false` to `is_trivia()`.

Reply:
There isn't one today, but I don't think that, by definition, there could _never_ be a trivia node. As such, in practice, this will always returned `false` today but would work in the future if we ever introduced a trivia node.